components/activecampaign/sources/contact-added-to-list/contact-added-to-list.js

```diff
@@ -6,9 +6,57 @@ module.exports = {
   name: "New Contact Added to List",
   key: "activecampaign-contact-added-to-list",
   description: "Emits an event each time a new contact is added to a list.",
-  version: "0.0.1",
+  version: "0.0.2",
+  props: {
+    ...common.props,
+    lists: { propDefinition: [activecampaign, "lists"] },
+  },
+  hooks: {
+    async activate() {
+      const sources =
+        this.sources.length > 0
+          ? this.sources
+          : this.activecampaign.getAllSources();
+      const hookIds = [];
+      const events = this.getEvents();
+      if (this.lists.length > 0) {
+        for (const list of this.lists) {
+          const { id } = await this.getHookData(
+            events,
+            this.http.endpoint,
+            sources,
+            list
+          );
+          hookIds.push(id);
+        }
+      } else {
+        const { id } = await this.getHookData(
+          events,
+          this.http.endpoint,
+          sources
+        );
+        hookIds.push(id);
+      }
+      this.db.set("hookIds", hookIds);
+    },
+    async deactivate() {
+      const hookIds = this.db.get("hookIds");
+      for (const id of hookIds) {
+        await this.activecampaign.deleteHook(id);
+      }
+    },
+  },
   methods: {
     ...common.methods,
+    async getHookData(events, endpoint, sources, list) {
+      const { webhook } = await this.activecampaign.createHook(
+        events,
+        endpoint,
+        sources,
+        list
+      );
+      return webhook;
+    },
```

Comment:
We should remove this method, as it can be misleading.
`getHookData` implies that it's retrieving some value, when in fact what it does is create a webhook.